web/src/server_events_dispatch.js

```diff
@@ -473,6 +474,7 @@ export function dispatch_normal_event(event) {
                     if (people.is_valid_bot_user(event.person.user_id)) {
                         settings_users.update_bot_data(event.person.user_id);
                     }
+                    navbar_alerts.maybe_show_empty_required_profile_fields_alert();
```

Comment:
This should probably be placed within this block within `user_events.update_person`. There's no reason to call that function except for custom profile field updates.

```
    if (Object.hasOwn(person, "custom_profile_field")) {                                                
        people.set_custom_profile_field_data(person.user_id, person.custom_profile_field);              
    }                                                                                                   
```

Reply:
Fixed in the recent commit.